tagstudio/src/qt/ts_qt.py

```diff
@@ -393,11 +395,27 @@ def start(self):
 				lib = self.settings.value("last_library")
 			self.splash.showMessage(f'Opening Library "{lib}"...', int(Qt.AlignmentFlag.AlignBottom|Qt.AlignmentFlag.AlignHCenter), QColor('#9782ff'))
 			self.open_library(lib)
+		
+		def refresh_library():
+			self.lib.refresh_dir()
+			self.preview_panel.update_widgets()
+			self.filter_items()
+
+		class IsDirModifiedHandler(FileSystemEventHandler):
+			def on_any_event(self, event):
+				if not event.is_directory:
+					refresh_library()
+		
+		observer = Observer()
+		observer.schedule(IsDirModifiedHandler(), lib, recursive=True)
+		observer.start()
 
 		app.exec_()
 
-		self.shutdown()
+		observer.stop()
+		observer.join()
```

Comment:
Also, maybe instead of setting it as a method variable, make it accessible through self.observer 
Then, you could move the observer.stop() and observer.join() into self.shutdown

Reply:
Thanks for your input, I'll follow your advice with future changes